java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/project/BrokenReferencesModel.java

```diff
@@ -0,0 +1,318 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.netbeans.modules.java.lsp.server.project;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.WeakHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.swing.AbstractListModel;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import org.netbeans.api.annotations.common.NonNull;
+import org.netbeans.api.project.Project;
+import org.netbeans.api.project.ProjectManager;
+import org.netbeans.api.project.ProjectUtils;
+import org.openide.util.ChangeSupport;
+import org.netbeans.spi.project.ui.ProjectProblemsProvider;
+import org.netbeans.spi.project.ui.ProjectProblemsProvider.ProjectProblem;
+import org.openide.util.Mutex;
+import org.openide.util.NbBundle;
+import org.openide.util.WeakListeners;
+
+/**
+ * This is a copy of broken references model from Project UI API, slightly improved.
+ * 
+ */
+public final class BrokenReferencesModel extends AbstractListModel implements PropertyChangeListener, ChangeListener {
```

Comment:
The alternative might be to use `Node` instead of `javax.swing.AbstractListModel` and expose the hierarchy from Project UI API module rather than copying the code.

Reply:
Do you mean to use Node / Children as a model, ProjectProblem as a service in its Lookup ?

Reply:
Btw. if you had `Node` representing the problems, you could show it in the logical project view under the project node... that might be nicer UI than a sequence of dialogs...

Reply:
Hm, as an alternative - possibly. I did not thought about chaning NB IDE UI by this PR ... and I still need to think this through - the error messages might be quite long -> display as node's description (tooltip):  does not seem as good usability pattern.
The API `ProjectProblems.alert()` is used & useful (it is called after project open).

Reply:
I'll keep the data model as it is now -- it will not escape the implementation package, and although types are in Swing lib, they do not make threading manipulations.